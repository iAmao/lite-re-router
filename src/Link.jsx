import React from 'react';
import PropTypes from 'prop-types';


const Link = ({ children, href,...props}, context) => {
  return (
    <a {...props} onClick={() => context.location.push(href)}>
      {children}
    </a>
  );
};

Link.contextTypes = {
  location: PropTypes.shape({
    push: PropTypes.func,
    path: PropTypes.string,
  })
};

Link.propTypes = {
  href: PropTypes.string.isRequired,
  children: PropTypes.node.isRequired
};

export default Link;
