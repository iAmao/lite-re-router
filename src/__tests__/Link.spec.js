import React from 'react';
import renderer from 'react-test-renderer';

import { shallow } from 'enzyme';

import Link from '../Link';


describe('Link', () => {
    describe('Snapshot Test', () => {
        it('renders a link', () => {
            const tree = renderer.create(
                <Link href="/button">Button</Link>
            ).toJSON();
            expect(tree).toMatchSnapshot();
        });
    });

    describe('Unit Test', () => {
        it('should change context on click', () => {
            let count = 0;
            const wrapper = shallow(Link(
                { children: 'Button', href: '/button' },
                { location: { push: () => count++}}
            ));

            wrapper.simulate('click');

            expect(count).toEqual(1);
        })
    });
});
