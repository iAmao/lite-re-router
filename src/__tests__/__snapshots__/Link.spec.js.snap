// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link Snapshot Test renders a link 1`] = `
<a
  onClick={[Function]}
>
  Button
</a>
`;
